src/Symfony/Component/Validator/Tests/Mapping/Factory/LazyLoadingMetadataFactoryTest.php

```diff
@@ -168,6 +170,24 @@ public function testMetadataCacheWithRuntimeConstraint()
 
         $metadata = $factory->getMetadataFor(self::CLASS_NAME);
     }
+
+    public function testNonClassNameStringValues()
+    {
+        $testedValue = 'error@example.com';
+        $loader = $this->getMockBuilder('Symfony\Component\Validator\Mapping\Loader\LoaderInterface')->getMock();
+        $cache = $this->getMockBuilder('Symfony\Component\Validator\Mapping\Cache\CacheInterface')->getMock();
+        $factory = new LazyLoadingMetadataFactory($loader, $cache);
+        $cache
+            ->method('read')
+            ->with($testedValue)
+            ->willThrowException(new InvalidArgumentException(sprintf('Cache key "%s" contains reserved characters {}()/\@:', $testedValue)));
+        $this->expectException(NoSuchMetadataException::class);
+        try {
+            $factory->getMetadataFor($testedValue);
+        } catch (InvalidArgumentException $exception) {
```

Comment:
no need for this then. We expect `NoSuchMetadataException::class` and if not the test just fails